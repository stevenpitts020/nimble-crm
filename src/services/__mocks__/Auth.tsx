export const successResponse = {
  type: 'MAGIC_LINK',
  message: 'Login-link sent to [test@wmail.com].',
  token:
    'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzUxMiJ9.eyJpZCI6MTAsImVtYWlsIjoiZGVtb3VzZXJAbmltYmxlZmkuY29tIiwiYXQiOjE1NzcxMDkwMjg1NDR9.tirQRQKltyi0ysCvfY9tn702ZYM7WBzDuRdq-13G5fLE95KJ7EkPIq95yAm6ty-9hbei6IKainh_1QZJVVypvw',
}
export const invalidResponse = {
  statusCode: 401,
  message: 'No authorization!',
}
export const errorResponse = {
  statusCode: 400,
  message:
    'The minimal required parameters for this endpoint were not met. "password" is not allowed to be empty',
}

export const recoverPassSuccess = {
  statusCode: 204,
  message: '',
}

export const recoverPassNotFound = {
  statusCode: 404,
  message: 'Could not find an user with that email.',
}

export const recoverPassIsExpired = {
  statusCode: 404,
  message: 'Could not change your password. please request a new code.',
}

export const passwordChangedSuccess = {
  id: 6,
  email: 'vlad@wearesingular.com',
  firstName: 'Vladislav',
  lastName: 'Sorokin',
  gender: 'male',
  createdAt: '2020-04-17T15:34:25.165Z',
  institution: {
    id: 'centralbankonline',
    name: 'Central Bank',
    domain: 'centralbankonline.com',
    logoUri: {
      default:
        'https://nimble-assets-dev.s3.us-east-2.amazonaws.com/logo/76ea47f7-0335-4f21-b8a4-6f6ec01ee767.png',
    },
    backgroundImageUri: {
      default:
        'https://nimble-assets-dev.s3.us-east-2.amazonaws.com/background/46245139-d5fb-4767-8a26-049bb8437a54.png',
    },
  },
  accounts: [
    {
      strategy: 'local',
      remoteId: null,
      createdAt: '2020-04-17T15:34:25.178Z',
    },
  ],
}
