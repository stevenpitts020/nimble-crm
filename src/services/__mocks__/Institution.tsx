export const successResponse = {
  id: 'singular',
  name: 'Singular',
  domain: 'wearesingular.com',
  backgroundImageUri: {
    default: 'http://wearesingular.com/img/intro-bg.jpg',
  },
  logoUri: {
    default: 'https://wearesingular.com/img/apple-touch-icon.png',
  },
  templateApprove:
    '{accountNumber} {amount} {routingNumber} {bankName} {name} {product} {employee}',
  templateDecline: '{bankName} {name} {product} {employee}',
}

export const invalidResponse = {
  statusCode: 401,
  message: 'Request failed with status code 401',
}
