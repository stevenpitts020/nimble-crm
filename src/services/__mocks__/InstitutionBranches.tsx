export const successResponse = [
  {
    id: 'ff6dff47-e159-4062-b886-bb2e092db0ab',
    name: 'Sioux City Morningside',
    externalId: '52',
  },
  {
    id: 'ed5de51b-3e81-480f-8af6-2de558b41b65',
    name: 'Sioux City Hamilton',
    externalId: '51',
  },
  {
    id: 'e5d1387f-14ea-489c-bcfb-b123e10e9f05',
    name: 'Sioux Empire - 33rd & Minnesota',
    externalId: '40',
  },
  {
    id: 'e19469f2-bd1f-480d-bb3f-929712685c43',
    name: 'Brookings',
    externalId: '47',
  },
  {
    id: 'de1cd5f7-ee12-4607-8d0b-f902b7fc5b43',
    name: 'Storm Lake Plaza',
    externalId: '6',
  },
  {
    id: 'da309f62-7a20-4332-b292-4dab45b220d0',
    name: 'Storm Lake Main',
    externalId: '1',
  },
  {
    id: 'af22ae8b-6e0b-4fc2-b068-a9ec68a1b944',
    name: 'Dakota Dunes',
    externalId: '53',
  },
  {
    id: 'a565b12a-534f-43da-b44c-b16e7d96719f',
    name: 'Westown',
    externalId: '70',
  },
  {
    id: '9135b395-5300-4f07-939f-4e5ef3cc7b67',
    name: 'Spirit Lake',
    externalId: '60',
  },
  {
    id: '8aea7f30-58b7-4bc0-88ab-fc7c56ad0c31',
    name: 'Cherokee North',
    externalId: '31',
  },
  {
    id: '8ac30f46-2dbe-4438-9862-b405188df58e',
    name: 'Ingersoll',
    externalId: '81',
  },
  {
    id: '7a2abe29-5925-4d2d-9818-c7fae6eedd9f',
    name: 'Ankeny',
    externalId: '71',
  },
  {
    id: '685d9220-837c-4acf-ab51-71bc7f930159',
    name: 'Waukee',
    externalId: '78',
  },
  {
    id: '595b3fbd-22cc-44fb-8913-f7d25ee9d842',
    name: 'Mills Civic',
    externalId: '73',
  },
  {
    id: '4a762a6d-699d-4a89-8516-ded0acfe5f62',
    name: 'Sioux Empire - Western',
    externalId: '42',
  },
  {
    id: '3ac5bb7a-1cac-4b9f-afac-0961c98ddf28',
    name: 'Storm Lake North',
    externalId: '2',
  },
  {
    id: '37aa3e02-f67c-4d8d-b492-8998a7f525fd',
    name: 'Sioux City Downtown',
    externalId: '50',
  },
  {
    id: '32df8a73-3779-4c80-960f-8ef0534f42bf',
    name: 'Highland Park',
    externalId: '82',
  },
  {
    id: '28ae1b8c-68f8-424b-9d91-0ecf37f8a7ed',
    name: 'Urbandale - 100th',
    externalId: '74',
  },
  {
    id: '27c50932-285f-49ac-80b3-621e5c64b795',
    name: 'Sioux Empire - W 12th',
    externalId: '41',
  },
  {
    id: '24901c7f-bfcd-4175-87cb-985e95b79d93',
    name: 'Urbandale - 86th',
    externalId: '80',
  },
  {
    id: '1c835549-4445-4a7b-9f40-df484bb8d773',
    name: 'Storm Lake 5th',
    externalId: '5',
  },
  {
    id: '18a80224-d12a-461d-9038-42fe0e341934',
    name: 'Cherokee',
    externalId: '30',
  },
  {
    id: '13ceee2f-4bb8-4eb1-904d-4625ce1c6b29',
    name: 'Drake',
    externalId: '75',
  },
]
