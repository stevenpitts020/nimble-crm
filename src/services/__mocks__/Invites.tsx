export const successResponse = {
  id: '8f862362-d703-4228-b4e7-2681721fee42',
}

export const invalidResponse = {
  statusCode: 400,
  message: 'Request failed with status code 400',
}
