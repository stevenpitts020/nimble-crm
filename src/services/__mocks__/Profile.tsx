export const successResponse = {
  id: 0,
  email: 'someuser@wearesingular.pt',
  firstName: 'John',
  lastName: 'Goodman',
  gender: 'male',
  profile: {
    birthDate: '2019-12-23T15:32:45.397Z',
    avatar: {
      default: 'string',
    },
  },
  accounts: [
    {
      strategy: 'string',
      remoteId: 'string',
      createdAt: '2019-12-23T15:32:45.397Z',
    },
  ],
  institution: {
    id: 'singular',
    name: 'Singular',
    domain: 'wearesingular.com',
    backgroundImageUri: {
      default:
        'https://nimble-assets-dev.s3.us-east-2.amazonaws.com/background/background.jpg',
    },
    logoUri: {
      default:
        'https://nimble-assets-dev.s3.us-east-2.amazonaws.com/logo/company.jpg',
    },
    templateApprove:
      '{accountNumber} {amount} {routingNumber} {bankName} {name} {product} {employee}',
    templateDecline: '{bankName} {name} {product} {employee}',
  },
  createdAt: '2019-12-23T15:32:45.397Z',
}

export const invalidResponse = {
  statusCode: 401,
  message: 'Invalid token: no JWT token found',
}
