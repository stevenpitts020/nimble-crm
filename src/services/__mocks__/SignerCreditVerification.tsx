export const successResponse = [
  {
    id: '2e31d8c0-1226-4651-8a5d-4bd8aa454722',
    signerId: '2e31d8c0-1226-4651-8a5d-4bd8aa454722',
    completionCode: '0000',
    reference: '188504_1603905845',
    score: 744,
    reportDate: '10282020',
    report: {
      uri:
        'https://api.nimblefi.com/v1/documents/1052ab85-08da-4bb5-be00-9e94d282d310?token=8mAABig0OH9lja2um.8mAABig0OH9lja2um.8mAABig0OH9lja2um',
    },
    createdAt: '2020-09-16T11:32:38.558Z',
    updatedAt: '2020-09-16T11:32:38.558Z',
  },
]

export const invalidResponse = {
  statusCode: 404,
  message: 'The requested resource could not be found',
}
