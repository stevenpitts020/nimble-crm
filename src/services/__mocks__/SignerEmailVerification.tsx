export const successResponse = {
  statusCode: 204,
  message: 'Success',
}

export const invalidResponse = {
  statusCode: 404,
  message: 'The requested resource could not be found',
}
