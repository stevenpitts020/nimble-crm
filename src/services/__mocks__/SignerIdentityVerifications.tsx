export const successIdentityVerificationResponse = [
  {
    verification: 'Face Detection',
    category: 'Face',
    date: '2021-02-19T00:06:31.261Z',
    status: 'INVALID',
  },
  {
    verification: 'Face On Document Matched',
    category: 'Face',
    date: '2021-02-19T00:06:31.265Z',
    status: 'INVALID',
  },
  {
    verification: 'Originality',
    category: 'Document',
    date: '2021-02-19T00:06:31.268Z',
    status: 'INVALID',
  },
  {
    verification: 'Visibility',
    category: 'Document',
    date: '2021-02-19T00:06:31.271Z',
    status: 'INVALID',
  },
  {
    verification: 'Document Type',
    category: 'Document',
    date: '2021-02-19T00:06:31.274Z',
    status: 'VALID',
  },
  {
    verification: 'Issued Date',
    category: 'Document',
    date: '2021-02-19T00:06:31.277Z',
    status: 'INVALID',
  },
  {
    verification: 'Expiration Date',
    category: 'Document',
    date: '2021-02-19T00:06:31.280Z',
    status: 'INVALID',
  },
  {
    verification: 'Name',
    category: 'Document',
    date: '2021-02-19T00:06:31.283Z',
    status: 'VALID',
  },
  {
    verification: 'Country',
    category: 'Document',
    date: '2021-02-19T00:06:31.285Z',
    status: 'VALID',
  },
  {
    verification: 'Date Of Birth',
    category: 'Document',
    date: '2021-02-19T00:06:31.289Z',
    status: 'VALID',
  },
  {
    verification: 'Address',
    category: 'Document',
    date: '2021-02-19T00:06:31.292Z',
    status: 'INVALID',
  }
]

export const invalidResponse = {
  statusCode: 404,
  message: 'The requested resource could not be found',
}
