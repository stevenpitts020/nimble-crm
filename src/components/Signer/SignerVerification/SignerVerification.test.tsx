import React from 'react'
import { render } from '@testing-library/react'
import {
  SignerVerification,
  validCheck,
  docValid,
  complyAdvantageCheck,
} from './SignerVerification'
import { successResponse } from '../../../services/__mocks__/AccountRequest'

const renderApp = (props: any) => {
  return render(<SignerVerification {...props} />)
}

describe.skip('<SignerVerification />', () => {
  test('should render content', () => {
    const props = {
      identityVerification:
        successResponse.signers[0].identityVerificationResult,
      shuftiValidationDate: successResponse.createdAt,
      complyAdvantageChecks: {
        checkSanction: successResponse.signers[0].checkSanction,
        sanctionVerifiedAt: successResponse.signers[0].sanctionVerifiedAt,
        checkPoliticalExposure:
          successResponse.signers[0].checkPoliticalExposure,
        politicalExposureVerifiedAt:
          successResponse.signers[0].politicalExposureVerifiedAt,
        checkAdverseMedia: successResponse.signers[0].checkAdverseMedia,
        adverseMediaVerifiedAt:
          successResponse.signers[0].adverseMediaVerifiedAt,
        checkAntiMoneyLaundering:
          successResponse.signers[0].checkAntiMoneyLaundering,
        antiMoneyLaunderingVerifiedAt:
          successResponse.signers[0].antiMoneyLaunderingVerifiedAt,
      },
      agreement: {
        signerStatus: 'PENDING',
        accountRequestStatus: 'PENDING',
      },
    }
    const { getByTestId } = renderApp(props)

    const role = getByTestId('ni-signer-verification')
    expect(role).toBeInTheDocument()
  })

  describe('validCheck', () => {
    // rendering variations on env require --updateSnapshot to be run on test runner instance
    test.skip('return ok and a date', () => {
      const result = validCheck(1, '12-03-2097')
      expect(result).toMatchSnapshot()
    })

    // rendering variations on env require --updateSnapshot to be run on test runner instance
    test.skip('return not ok and a date', () => {
      const result = validCheck(0, '12-03-2097')
      expect(result).toMatchSnapshot()
    })

    // rendering variations on env require --updateSnapshot to be run on test runner instance
    test.skip('return not ok with undefined and a date', () => {
      const result = validCheck(undefined, '12-03-2097')
      expect(result).toMatchSnapshot()
    })
  })

  describe('docValid', () => {
    test('return ok if successful', () => {
      const result = docValid({
        issue_date: 0,
        document_visibility: 0,
        document_must_not_be_expired: 1,
        document: 1,
        document_country: 0,
        selected_type: 0,
        document_proof: 1,
        name: 0,
        face_on_document_matched: 0,
      })
      expect(result).toEqual(1)
    })

    test('return 1 if undefined', () => {
      const result = docValid(undefined)
      expect(result).toEqual(0)
    })

    test('return 0 if failed', () => {
      const result = docValid({
        issue_date: 0,
        document_visibility: 0,
        document_must_not_be_expired: 1,
        document: 0,
        document_country: 0,
        selected_type: 0,
        document_proof: 0,
        name: 0,
        face_on_document_matched: 0,
      })
      expect(result).toEqual(0)
    })
  })

  describe('compyAdvantage check', () => {
    test('return ok and a date', () => {
      const result = complyAdvantageCheck(true, new Date())
      const { getByTestId } = render(result)
      const redIcon = getByTestId('failed')
      expect(redIcon).toBeInTheDocument()
    })

    test('return not ok and a date', () => {
      const result = complyAdvantageCheck(false, new Date())
      const { getByTestId } = render(result)
      const greenIcon = getByTestId('passed')
      expect(greenIcon).toBeInTheDocument()
    })

    test('return not ok with null and a date', () => {
      const result = complyAdvantageCheck(null, new Date())
      const { getByTestId } = render(result)
      const redIcon = getByTestId('failed')
      expect(redIcon).toBeInTheDocument()
    })
  })

  describe('contract signature check', () => {
    const props = {
      identityVerification:
        successResponse.signers[0].identityVerificationResult,
      shuftiValidationDate: successResponse.createdAt,
      complyAdvantageChecks: {
        checkSanction: successResponse.signers[0].checkSanction,
        sanctionVerifiedAt: successResponse.signers[0].sanctionVerifiedAt,
        checkPoliticalExposure:
          successResponse.signers[0].checkPoliticalExposure,
        politicalExposureVerifiedAt:
          successResponse.signers[0].politicalExposureVerifiedAt,
        checkAdverseMedia: successResponse.signers[0].checkAdverseMedia,
        adverseMediaVerifiedAt:
          successResponse.signers[0].adverseMediaVerifiedAt,
        checkAntiMoneyLaundering:
          successResponse.signers[0].checkAntiMoneyLaundering,
        antiMoneyLaunderingVerifiedAt:
          successResponse.signers[0].antiMoneyLaunderingVerifiedAt,
      },
      agreement: {
        signerStatus: 'PENDING',
        accountRequestStatus: 'PENDING',
      },
    }
    test('return pending signature', () => {
      const { getByText, getByTestId } = renderApp(props)
      const text = getByText(/Pending Signature/)
      const redIcon = getByTestId('pending-contract')
      expect(text).toBeInTheDocument()
      expect(redIcon).toBeInTheDocument()
    })
    test('return signed waiting for other signers', () => {
      const completedprops = {
        ...props,
        agreement: {
          signerStatus: 'PENDING',
          accountRequestStatus: 'INCOMPLETE',
        },
      }
      const { getByText, getByTestId } = renderApp(completedprops)
      const text = getByText(/Waiting for other signers/)
      const redIcon = getByTestId('pending-contract')
      expect(text).toBeInTheDocument()
      expect(redIcon).toBeInTheDocument()
    })
    test('return signed and green icon', () => {
      const completedprops = {
        ...props,
        agreement: {
          signerStatus: 'SIGNED',
          accountRequestStatus: 'SIGNED',
        },
      }
      const { getByText, getByTestId } = renderApp(completedprops)
      const text = getByText(/Signed/)
      const greenIcon = getByTestId('signed-contract')
      expect(text).toBeInTheDocument()
      expect(greenIcon).toBeInTheDocument()
    })
  })
})
