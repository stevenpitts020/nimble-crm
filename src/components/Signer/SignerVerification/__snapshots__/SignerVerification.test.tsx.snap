// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignerVerification /> validCheck return not ok and a date 1`] = `
<React.Fragment>
  <AlertTriangleIcon
    style={
      Object {
        "fontSize": "18px",
        "paddingRight": "6px",
      }
    }
  />
   
  12-03-2097
</React.Fragment>
`;

exports[`<SignerVerification /> validCheck return not ok with undefined and a date 1`] = `
<React.Fragment>
  <AlertTriangleIcon
    style={
      Object {
        "fontSize": "18px",
        "paddingRight": "6px",
      }
    }
  />
   
  12-03-2097
</React.Fragment>
`;

exports[`<SignerVerification /> validCheck return ok and a date 1`] = `
<React.Fragment>
  <CheckCircleIcon
    style={
      Object {
        "fontSize": "18px",
        "paddingRight": "6px",
      }
    }
  />
   
  12-03-2097
</React.Fragment>
`;
